### PR TITLE
samples: added STeP mobilie kinematic simulation to illustrate a multicore sample

### PR DESCRIPTION
**Pull request by uLipe**

### CODE REVIEW

**Comment by uLipe** [comment]
I forgot to tag 🤦 

@microbuilder PTAL.

File: samples/mobile_robot_kinematic_server/remote/src/kinematic_node.c [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    tan_rpm = tangential_vel / tx_payload->kinematics.wheel_radius;;
+
+    tx_payload->joints.joint_speed_rpm[0] = x_rpm - y_rpm - tan_rpm;
+	tx_payload->joints.joint_speed_rpm[1] = x_rpm + y_rpm + tan_rpm;
+    tx_payload->joints.joint_speed_rpm[2] = x_rpm + y_rpm - tan_rpm;
+    tx_payload->joints.joint_speed_rpm[3] = x_rpm - y_rpm + tan_rpm;
+
+	/*pushes robot data back to the transmitting core to compare the results */
+	return rpmsg_service_send(ep_id, tx_payload, sizeof(struct mobile_robot_node_payload));
+}
```
Comment (microbuilder) [inline samples/mobile_robot_kinematic_server/remote/src/kinematic_node.c:104]: nit: Whitespace

File: samples/mobile_robot_kinematic_server/src/kinematic_node.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+																	rx_payload.body.linear_velocity_y,
+																	rx_payload.body.heading);
+	printk("Output joint RPMs from Remote Core: %f, %f, %f, %f [rad/min] \n", rx_payload.joints.joint_speed_rpm[0],
+													rx_payload.joints.joint_speed_rpm[1],
+													rx_payload.joints.joint_speed_rpm[2],
+													rx_payload.joints.joint_speed_rpm[3]);
+	printk("=====================================================================\n");
+
+	return RPMSG_SUCCESS;
+}
```
Comment (microbuilder) [inline samples/mobile_robot_kinematic_server/src/kinematic_node.c:52]: nit: whitespace

**Review by microbuilder — CHANGES_REQUESTED** [review]

**Comment by microbuilder** [comment]
@uLipe What version of Zephyr is this being built against, and what QEMU version are you using? For some reason, when I build and run via:

```bash
$ west build  -p always -b mps2_an521  modules/lib/step/samples/mobile_robot_kinematic_server
$ west build -t run
```

I can't enter anyting in the Zephyr shell, which is a first for me.

```bash
$ qemu-system-arm --version
QEMU emulator version 7.0.0 (v7.0.0-dirty)
Copyright (c) 2003-2022 Fabrice Bellard and the QEMU Project developers

$ git describe
zephyr-v3.2.0-2456-g7c9d7922116
```

**Comment by uLipe** [comment]
Just update the README.

**Comment by uLipe** [comment]
@microbuilder , depending on version of Zephyr it is necessary to send a request for the remote processor before wakeup, it is also happens on the rpmsg demo, and only for the QEMU targets, I moved the wakeup request position and it should work now regardless the zephyr version .

**Pull request merged**